process/transaction/baseProcess.go

```diff
@@ -214,3 +217,32 @@ func (txProc *txProcessor) VerifyTransaction(tx *transaction.Transaction) error
 
 	return txProc.checkTxValues(tx, senderAccount, receiverAccount, false)
 }
+
+func (txProc *baseTxProcessor) verifyGuardian(tx *transaction.Transaction, account state.UserAccountHandler) error {
+	if check.IfNil(account) {
+		return nil
+	}
+	if !account.IsFrozen() {
+		return nil
+	}
+
+	if !txProc.txVersionChecker.IsGuardedTransaction(tx) {
+		return nil
+	}
+
+	acc, ok := account.(vmcommon.UserAccountHandler)
+	if !ok {
+		return process.ErrWrongTypeAssertion
+	}
+
+	guardian, err := txProc.guardianChecker.GetActiveGuardian(acc)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(guardian, tx.GuardianAddr) {
```

Comment:
the "SetGuardian" tx should bypass this check in the next PR?

Reply:
Here we just make sure at the processing time (right before the tx execution) that the guardian for which the signature was verified when intercepting the transaction is still the active guardian, so this should not be bypassed.